### PR TITLE
Add unit tests for OnButtonEvent&OnButtonPress

### PR DESCRIPTION
**Pull request by OHerasym**
PR Changes:
-Add unit tests according CRQ for OnButtonEvent
-Add unit tests according CRQ for OnButtonPress

Related: [APPLINK-25473](https://adc.luxoft.com/jira/browse/APPLINK-25473)

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @dtrunov , @anosach-luxoft , @LuxoftAKutsan ,@VVeremjova, @orhan-mehmedov

File: src/components/application_manager/test/commands/mobile/on_button_press_notification_test.cc [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    command_->Run();
+  }
+
+  NotificationPtr command_;
+};
+
+TEST_F(OnButtonPressNotificationTest,
+       DISABLED_Run_SendOnButtonPress_in_FULL_Case1_SUCCESS) {
+  CheckExpectations(false, HMILevel::HMI_FULL, 10, 8, true, true);
```
Comment (dtrunov) [inline src/components/application_manager/test/commands/mobile/on_button_press_notification_test.cc]: @OHerasym Magic number.

**Comment by dtrunov** [comment]
@OHerasym Reviewed

File: src/components/application_manager/test/commands/mobile/on_button_press_notification_test.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/application_manager/test/commands/mobile/on_button_press_notification_test.cc]: @dtrunov , 9ed629a, Done

**Comment by orhan-mehmedov** [comment]
@OHerasym - why all tests are disabled?

**Comment by OHerasym** [comment]
@orhan-mehmedov , CRQ implementation will be in master branch, and tests are send to develop, so they will be enabled after merging master to develop

**Comment by VVeremjova** [comment]
@OHerasym Reviewed

**Comment by anosach-luxoft** [comment]
Reviewed

**Comment by orhan-mehmedov** [comment]
@OHerasym Reviewed

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+using ::testing::_;
+using ::testing::Return;
+using namespace mobile_apis;
+using am::commands::MessageSharedPtr;
+using am::commands::mobile::OnButtonEventNotification;
+
+typedef ::utils::SharedPtr<OnButtonEventNotification> NotificationPtr;
+
+namespace {
+const int app_id1 = 10;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @OHerasym Looks like it is constants, please use appropriate style

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                         const size_t app_id2,
+                         const bool is_full_screen = false,
+                         const bool is_send_message_expected = false) {
+    MockAppPtr mock_app(CreateMockApp());
+    MockAppPtr subscribed(CreateMockApp());
+    std::vector<ApplicationSharedPtr> subscribedApps;
+    subscribedApps.push_back(subscribed);
+
+    if (is_app_id_valid) {
+      EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(mock_app));
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @OHerasym I think you should care about argument to application

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        .WillOnce(Return(subscribedApps));
+    EXPECT_CALL(*subscribed, hmi_level()).WillRepeatedly(Return(app_hmi_level));
+    EXPECT_CALL(*subscribed, app_id()).WillRepeatedly(Return(app_id1));
+    EXPECT_CALL(*mock_app, app_id()).WillRepeatedly(Return(app_id2));
+    if (is_full_screen) {
+      EXPECT_CALL(*subscribed, IsFullscreen()).WillOnce(Return(true));
+    }
+
+    EXPECT_CALL(app_mngr_, SendMessageToMobile(_, _))
+        .Times(is_send_message_expected);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @OHerasym TImes(bool)? Are you sure that is it is correct? At least it is not clear

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+    command_->Run();
+  }
+
+  NotificationPtr command_;
+};
+
+TEST_F(OnButtonEventNotificationTest,
+       DISABLED_Run_SendOnButtonEvent_in_FULL_Case1_SUCCESS) {
+  CheckExpectations(!app_id_valid,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @OHerasym Why did you created disabled tests?

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc (hunk 4 above)
Comment (OHerasym) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @LuxoftAKutsan , bool translates to int, bool true = > Times(1)

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc (hunk 5 above)
Comment (OHerasym) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @LuxoftAKutsan , CRQ implementation will be in master branch, and tests are send to develop, so they will be enabled after merging master to develop

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc (hunk 3 above)
Comment (OHerasym) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @LuxoftAKutsan , application() takes app_id argument, it's doesn't matter what is app_id, if I return expected mock_app.

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc (hunk 2 above)
Comment (OHerasym) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @LuxoftAKutsan , 51f30cd, Done

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc (hunk 4 above)
Comment (OHerasym) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @LuxoftAKutsan , 51f30cd, Done

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc (hunk 3 above)
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @OHerasym if you expect calling application i think you should check that caller code will call correct applicaiion

File: src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    command_->Run();
+  }
+
+  NotificationPtr command_;
+};
+
+TEST_F(OnButtonEventNotificationTest,
+       DISABLED_Run_SendOnButtonEvent_in_FULL_Case1_SUCCESS) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/on_button_event_notification_test.cc]: @OHerasym disabled tests

**Comment by LuxoftAKutsan** [comment]
Reviewed

**Pull request merged**